### PR TITLE
:sparkles: Add tracdelight AMP ad integration

### PR DESCRIPTION
**Pull request by thekorn**
The pull request contains implementation for the tracdelight ad-widgets for AMP Ads.
Reference to our closed PRs are #13956 #22550

### CODE REVIEW

**Comment by mrjoro** [comment]
Sorry for the delay in reviewing this @thekorn!

/cc @ampproject/wg-ads

**Comment by calebcordry** [comment]
@thekorn Thanks for contributing. When I try to run you example locally it doesn't seem to work. Looks like the request to `https://api.tracdelight.com/v1/widgets/false/products?accesskey=false` might not behave as expected. Would it be possible to return an example ad here? It makes it easier for us to have a working example so that we can ensure we don't break anything etc.

**Comment by calebcordry** [comment]
Also you might want to rebase since this pr has been out for a while (our fault). I had some weird errors before rebasing. Let me know if you need any assistance.

**Comment by thekorn** [comment]
thanks @calebcordry for the review, we will look into it and come back to you

**Comment by thekorn** [comment]
@calebcordry I've one question: I just pushed one change, and we got flagged `cla: no` again - altough I'm still the only author in the changeset. What is the reason for this?

**Comment by calebcordry** [comment]
@thekorn I see that the CLA was approved at one point, but when I look at our internal system it says your username is not registered. What do you see when you follow [this link](https://cla.developers.google.com/)?

**Comment by thekorn** [comment]
@calebcordry this is what I get on this page:
image

**Comment by calebcordry** [comment]
@thekorn in that screenshot I see your `@gmail` account, but the commits are signed from your `@tracdelight` account. I think that may be causing the issue. Could you confirm that your tracdelight email is also covered under the corporate CLA? Alternatively you could try [changing the author](https://help.github.com/en/articles/changing-author-info) of your commits to be your gmail address and see if that works.

**Comment by calebcordry** [comment]
Awesome, looks like you got the CLA all sorted out. I still don't see a working example, did this get lost in the CLA work?

Also, I don't see `renderStartImplemented` being called. Can you confirm that this is called in your JS? If not you might want to remove it from your config. We suggest implementing it, but if you have the flag as true without implementing, performance is actually worse. Let me know if you need any help. Thanks!

File: ads/tracdelight.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
```
Comment (calebcordry) [inline ads/tracdelight.js]: nit: 2019

**Comment by calebcordry** [comment]
image
Image of broken ad in case you wanted to see

**Comment by thekorn** [comment]
@calebcordry okay, thanks for your feedback - unfortunately the sample widget was not active anymore - I changed the widget id and now it's working properly.

2019-09-13 at 16 33

**Comment by thekorn** [comment]
Hi @calebcordry and @mrjoro 
just one question, as a kind of follow up, is there anything needed here from my side? anything I missed?
and what are the next steps here?
thanks for your help

File: ads/tracdelight.js (hunk 1 above)
Comment (calebcordry) [reply ads/tracdelight.js]: Can you change this please?

File: ads/tracdelight.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<!---
+Copyright 2018 The AMP HTML Authors. All Rights Reserved.
```
Comment (calebcordry) [inline ads/tracdelight.md]: nit: 2019

**Comment by calebcordry** [comment]
Sorry fo the delay. Just the year nits, and this should be ready.

File: ads/tracdelight.js (hunk 1 above)
Comment (thekorn) [reply ads/tracdelight.js]: @calebcordry done

**Comment by calebcordry** [comment]
Thanks, looks like a travis test flake. I just restarted, I will try to keep an eye on it, but let me know if you see it pass.

**Comment by calebcordry** [comment]
Passed :) thanks for contributing!

**Pull request merged**